### ISSUE TITLE
Retraining model on own data

### ISSUE THREAD

**Issue by gsrujana** [facebookresearch/sscd-copy-detection#6]
Hello,

what would be the way to fine-tune the sscd model on custom data? Is there an example on loading a checkpoint?

Thanks

**Comment by edpizzi** [comment]
To load a full lightning checkpoint (if you've run training from this codebase), you can load those full checkpoints using: `SSCD.load_from_checkpoint(checkpoint_path)`.

We don't export checkpoints in this format, since they are a bit fragile, and require pickling the whole SSCD class.

Our exported checkpoints can be loaded as follows:
```
    model = Model("TV_RESNET50", 512, 3.0)           
    model_state = torch.load("/path/to/sscd_disc_mixup.torchvision.pt", map_location=torch.device("cpu"))
    model.load_state_dict(model_state)
```

See the [Backbone](https://github.com/facebookresearch/sscd-copy-detection/blob/main/sscd/models/model.py#L21) enum for the model types, and match the checkpoint that you're loading.

Don't use the torchscript files for fine-tuning or training, only the Torchvision or Classy Vision models.

The [inference](https://github.com/facebookresearch/sscd-copy-detection/blob/main/sscd/lib/inference.py#L77-L84) library has examples of the two types of model loading, but it might be hard to follow because of `call_using_args`. (All `call_using_args` does is connect arguments from the argparse `args` structure to function parameters that have the same name as the argparse option.)

**Comment by gsrujana** [comment]
Thank you for your response! So, the suggestion is to replace https://github.com/facebookresearch/sscd-copy-detection/blob/main/sscd/train.py#L190 with the model load steps that you mentioned above? Also, is there any recommendation on freezing certain layers for retraining and how to make that change in your codebase with lr set for different layers? Thanks

**Comment by edpizzi** [comment]
Sorry for the delay. The location you indicated actually just constructs the model. My suggestion, with the caveat that I haven't tried it, would be to load weights for part or all of the model after the model is constructed, near here:

https://github.com/facebookresearch/sscd-copy-detection/blob/main/sscd/train.py#L463

For instance:
```
model = SSCD(...)
weights_filename = ...
weights = torch.load(weights_filename, ...)
model.load_state_dict(weights)
```

It's likely that the checkpoint file you have may need some editing to work with the constructed model. For instance, most torchvision models will contain a different fully-connected projection at the end of the network (eg. 2048 -> 1000 for ResNet50s trained for ImageNet classification). There are various ways to address this that I won't go into here (eg. removing some weights, then setting `strict=False`). Similarly, you may have to load weights to a module contained within the full SSCD model (eg. `model.model.backbone.load_state_dict(...)`).

I hope that helps, and good luck!
